tiny_dnn/layers/average_pooling_layer.h

```diff
@@ -141,19 +150,41 @@ class average_pooling_layer : public partial_connected_layer {
    * @param pool_size    [in] factor by which to downscale
    * @param stride       [in] interval at which to apply the filters to the
    *input
+   * @param ceil_mode    [in] when True, will use `ceil` instead of `floor` to
+   *compute the output shape
    **/
   average_pooling_layer(size_t in_width,
                         size_t in_height,
                         size_t in_channels,
                         size_t pool_size,
-                        size_t stride)
+                        size_t stride,
+                        bool ceil_mode = false)
     : average_pooling_layer(in_width,
                             in_height,
                             in_channels,
                             pool_size,
                             (in_height == 1 ? 1 : pool_size),
                             stride,
                             stride,
+                            ceil_mode,
+                            padding::valid) {}
+
+  average_pooling_layer(size_t in_width,
+                        size_t in_height,
+                        size_t in_channels,
+                        size_t pool_size_x,
+                        size_t pool_size_y,
+                        size_t stride_x,
+                        size_t stride_y,
```

Comment:
I'm not really sure why this constructor does not have `ceil_mode` argument.
Would it be a problem?